larq_compute_engine/core/bconv2d_output_transform.h

```diff
@@ -15,10 +15,39 @@ namespace core {
 
 using compute_engine::core::TBitpacked;
 
+// Clamp an int32 value to int8 range
+std::int8_t saturate(std::int32_t x) {
+#ifdef __thumb__
+  std::int8_t y;
+  asm("ssat %[y], 8, %[x]\n" : [ y ] "=r"(y) : [ x ] "r"(x));
+  return y;
+#else
```

Comment:
This is valid non-thumb Arm code too, right?

```suggestion
#ifdef __arm__
  std::int8_t y;
  asm("ssat %[y], 8, %[x]\n" : [ y ] "=r"(y) : [ x ] "r"(x));
  return y;
#else
```